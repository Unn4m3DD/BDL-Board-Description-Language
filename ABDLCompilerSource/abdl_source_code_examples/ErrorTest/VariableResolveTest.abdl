main:
    let v1 = 0;
    let v2 = "";
    let v3 = [0, 0];

    if 1 < 2 then
        let v2 = 0;
        v3 = v1 - v2; // should print an error
        let v3 : int;
        v3 = v1 - v2;
        let v4 = 2;
    end if;

    print(v4); // should print 2 errors
    for i from 0 to width do
        i = 2;
        let a = i;
        a = 2;
    done;

    i = 2; //should print error

    not_defined_function();
end main;