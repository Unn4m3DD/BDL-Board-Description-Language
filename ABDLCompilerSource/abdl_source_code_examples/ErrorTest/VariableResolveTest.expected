Variable and attribution types mismatch (8:8): v3 (point) and v1-v2 (int)
Undefined variable (14:10): v4
Invalid call to print statement (14:4)
Variable not declared (21:4): i
Function not defined (23:4): not_defined_function